x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StartTrainedModelDeploymentAction.java

```diff
@@ -185,18 +252,30 @@ public static TaskParams fromXContent(XContentParser parser) {
 
         private final String modelId;
         private final long modelBytes;
+        private final int inferenceThreads;
+        private final int modelThreads;
 
-        public TaskParams(String modelId, long modelBytes) {
+        public TaskParams(String modelId, long modelBytes, @Nullable Integer inferenceThreads, @Nullable Integer modelThreads) {
```

Comment:
```suggestion
        public TaskParams(String modelId, long modelBytes, int inferenceThreads, int modelThreads) {
```
Or you get NPE on the unboxing in `this.inferenceThreads = inferenceThreads;`